src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,166 @@
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+import java.time.LocalDate;
+
+/**
+ * Handler to load tasks from the file and save tasks into the file
+ */
+public class Storage {
+
+    private String filePath;
+
+    /**
+     * Constructs a Storage object
+     * @param filePath Location of the file to save the tasks into
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+        ensureFolderExists();
+        ensureFileExists();
+    }
+
+    /**
+     * Ensures that the given folder of the filePath exists, and creates the folder if it doesn't
+     */
+    private void ensureFolderExists() {
+        File folder = new File(filePath).getParentFile();
+        if (!folder.exists()) {
+            folder.mkdirs();
+        }
+    }
+
+    /**
+     * Ensures that the given file of the filePath exists, and creates file if it doesn't
+     */
+    private void ensureFileExists() {
+        File file = new File(filePath);
+        if (!file.exists()) {
+            try {
+                file.createNewFile();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    /**
+     * Loads tasks from the file
+     * @return List of tasks
+     */
+    public List<Task> loadTasks() {
+        List<Task> tasks = new ArrayList<>();
+
+        try {
+            File file = new File(filePath);
+            if (!file.exists()) {
+                file.createNewFile();
+            }
+
+            if (file.length() == 0) {
+                // File is empty, return an empty list
+                return tasks;
+            }
+
+            try (Scanner scanner = new Scanner(file)) {
+                while (scanner.hasNextLine()) {
+                    String line = scanner.nextLine();
+//                    System.out.println("Line: " + line); // Add this line for debugging
+                    String[] parts = line.split(" \\| ");
+
+                    if (parts.length >= 3) { // Ensure there are enough parts to proceed
+                        String type = parts[0];
+                        String isDone = parts[1];
+                        String description = parts[2];
+
+                        switch (type) {
+                            case "T":
+                                tasks.add(new ToDo(description, isDone));
+                                break;
+                            case "D":
+                                tasks.add(parseDeadline(parts, isDone));
+                                break;
+                            case "E":
+                                tasks.add(parseEvent(parts, isDone));
+                                break;
+                        }
+                    } else {
+                        System.out.println("Skipping line with insufficient parts: " + line);
+                    }
+                }
+            }
```

Comment:
perhaps can improve the readability of the code in this part by avoiding deep nesting?